### PR TITLE
Add a new date format used by the new API SDK

### PR DESCRIPTION
**Pull request by josvazg**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

**Comment by josvazg** [comment]
@helderjs did you find this format before in the previous migrations? I wonder if we had something else to avoid this.

**Comment by helderjs** [comment]
> @helderjs did you find this format before in the previous migrations? I wonder if we had something else to avoid this.

No, I haven't spotted this before. Where did you spot this? all endpoints or someone specific? cc @josvazg

**Comment by josvazg** [comment]
> > @helderjs did you find this format before in the previous migrations? I wonder if we had something else to avoid this.
> 
> No, I haven't spotted this before. Where did you spot this? all endpoints or someone specific? cc @josvazg

I found errors for this format in the PoC #1543 that upgrades the db user.

**Review by helderjs — CHANGES_REQUESTED** [review]
@josvazg This PR is not solving any issue, and I think it can be closed

File: internal/timeutil/timeutil.go [hunk 1]
```diff
@@ -5,7 +5,12 @@ import "time"
 // ParseISO8601 parses string as ISO8601 date. Note that the format allows different timezones, also time is optional,
 // so we have to list all possible formats
 func ParseISO8601(dateTime string) (time.Time, error) {
-	parse, err := time.Parse("2006-01-02T15:04:05-07", dateTime)
+	// Datetime format used by the new SDK
+	parse, err := time.Parse("2006-01-02 15:04:05 +0000 UTC", dateTime)
```
[the remark this answers is not in the document]
Comment (josvazg) [reply internal/timeutil/timeutil.go:9]: You are right!
I copied that [code from here](https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1543/files#diff-733b0371c07644c49998b08f8fbbdfd0827a48cf7c717754a972296f3b289f24L237) kind of bindly, I should have just cleaned it up instead.

This is not needed, as you say. Will fix the PoC instead.

**Pull request closed without merging**